fennel/datasets/test_schema_validator.py

```diff
@@ -601,9 +604,180 @@ class F:
             @pipeline(version=1)
             @inputs(A, E)
             def pipeline_join(cls, a: Dataset, e: Dataset):
-                return a.left_join(e, on=["a1"])
+                return a.join(e, how="left", on=["a1"])
 
     assert (
         str(e.value)
         == """Key field a1 has type str in left schema but type int in right schema."""
     )
+
+
+# dedup is not supported on keyed datasets
+def test_dedup_ds_with_key_fails():
+    with pytest.raises(ValueError) as e:
+
+        @meta(owner="abhay@fennel.ai")
+        @dataset
+        class MovieStats:
+            movie: str = field(key=True)
+            rating: float
+            revenue: int
+            t: datetime
+
+            @pipeline(version=1)
+            @inputs(MovieRating)
+            def pipeline_dedup(cls, rating: Dataset):
+                return rating.dedup(by=[MovieRating.movie])
+
+    assert (
+        str(e.value)
+        == """invalid dedup: input schema '[Dataset:MovieRating]' has key columns"""
```

Comment:
Fyi client follows the pattern that the user info is under ` not ' . Check other error messages for examples

Reply:
Ping on this